File: MPI_Example/random_walk.cpp
#include<iostream>
#include<random>
#include<vector>
#include<time>
#include"random_walker.h"

int main(int argc,char* argv[])
{

	//Prepare parameters
	int t_max=100;
	int n_max=10000;
	std::normal_distribution<double> normal(0,1);
	
	//Initialize random walker and trajectory
	std::vector<double> x(t_max);
	std::vector<double> y(t_max);
	x[0]=0,	y[0]=0;
	int seed=0;
	random_walker rw(x[0],y[0],seed);	

	//Ensemble loop
	clock_t start=clock();
	for(int i=0;i<n_max;i++)
	{
		rw.reset(x[0],y[0]);		

		//Time step loop
		for(int t=1;t<t_max;t++)
		{
			rw.step(normal);
			x[t]=rw.x();
			y[t]=rw.y();
		
		}
	}
	clock_t end=clock();

	std::cout<<n_max<<" random walks - "
		<<(double)(end-start)/CLOCKS_PER_SEC<<" seconds"<<std::endl;
	return 0;
}

File: MPI_Example/random_walk.py
import numpy as np
import time
import matplotlib.pyplot as plt
from random_walker import random_walker

tmax=1000
nmax=10000

start=time.time()
x_tr=[]
y_tr=[]
rw=random_walker(tmax,seed=1)
for i in range(nmax):
	rw.reset()
	rw.run()
	x,y=rw.get_trajectory()	
	x_tr.append(x)
	y_tr.append(y)

end=time.time()
print("Serial code takes ",end-start," seconds!")
'''
	Plotting routines
'''
import matplotlib.pyplot as plt
for i in range(10):
	plt.plot(x_tr[i],y_tr[i],lw=1)
plt.xlabel(r"$x$ position")
plt.ylabel(r"$y$ position")
#plt.savefig("Random_work_trajectories.png",dpi=300,bbox_inches='tight')
plt.show()
